plugins/kubernetes-backend/src/service/KubernetesBuilder.ts

```diff
@@ -262,12 +271,20 @@ export class KubernetesBuilder {
     clusterSupplier: KubernetesClustersSupplier,
     catalogApi: CatalogApi,
     proxy: KubernetesProxy,
+    permissionApi: PermissionEvaluator,
   ): express.Router {
     const logger = this.env.logger;
     const router = Router();
-    router.use('/proxy', proxy.createRequestHandler());
+    router.use('/proxy', proxy.createRequestHandler(permissionApi));
     router.use(express.json());
-
+    router.use(
+      createPermissionIntegrationRouter({
+        resourceType: RESOURCE_TYPE_KUBERNETES_RESOURCE,
+        permissions: kubernetesPermissions,
+        rules: [],
+        getResources: _ => Promise.resolve([]),
```

Comment:
@mtlewis @vinzscam @HHogg thoughts on this integration?

I think what we'd really want to do here is to have some form of meaningful resource identifiers right? So that it's possible to apply more granular permissions for the proxied resources. Is it possible to add that afterwards and start with this? Or is it something we need to consider upfront

Reply:
#15150 has some ideas for refinements -- I think those might be in the spirit of what you're talking about here

Reply:
I would be very interested in hearing other folks opinion on this. We decided to go with a barebones route that lists all the permissions available to the plugin at this route and the only permission that we introduce with this PR is of the type[BasicPermission](https://backstage.io/docs/reference/plugin-permission-common.basicpermission#__docusaurus). If in the future we were concerned with adding something more granular we would add support for a [ResourcePermission](https://backstage.io/docs/reference/plugin-permission-common.resourcepermission) and we could then probably concern ourselves with granular permissions allowed via [conditionalPolicyDecisions](https://backstage.io/docs/reference/plugin-permission-common.conditionalpolicydecision#__docusaurus). Anyways Im open to suggestions because this route was just something that we added along with the feature and is not required for the suggested changes to work.

Reply:
yeah this is something we are discussing these days and it's more likely happen that we would need to relax a bit the options accepted by `createPermissionIntegrationRouter`. In your case you shouldn't provide any `getResource` and `resourceType`, since your plugin doesn't provide any conditional decisions at the moment.
We'll get back asap!

Reply:
👋  as @vinzscam said, it's something we'd came across in the last couple of days and were planning to send up an API change for it.... here it is! #16333

Reply:
Awesome @HHogg i saw that your PR got merged. So I will take a look and see what changes were added. I take it certain fields are now optional? If i want to simply use the route to list the permissions that are available to the plugin in general are there any recommended fields I should add to createPermissionIntegrationRouter ?

Reply:
since your plugin is not exposing any rules the only needed field is `permissions`, all the rest can be removed

Reply:
@Rugvip Removed the getResources field which is now an optional field for createPermissionIntegratationRouter. Were there any additional comments on this route? I believe this route is still doing what we expected of it which was to introduce the ability to list all permissions available to the Kubernetes plugin at this local route http://localhost:7007/api/kubernetes/.well-known/backstage/permissions/metadata

Reply:
Thanks @vinzscam. I removed the unnecessary fields.